pkg/optimization/substat_details.go

```diff
@@ -0,0 +1,551 @@
+	var (
+		charDebug []string
+		opDebug   []string
+	)
+
+	for idxChar, char := range stats.charProfilesERBaseline {
+		charDebug = stats.findOptimalERforChar(idxChar, char, tolMean, tolSD)
+		opDebug = append(opDebug, charDebug...)
+	}
+
+	// Need a separate optimization routine for strong battery characters (currently Raiden only, maybe EMC?)
+	// Need to set all other character's ER substats at final value, then see added benefit from ER for global battery chars
+	for i, char := range stats.charProfilesERBaseline {
+		stats.charProfilesERBaseline[i].Stats[core.ER] = stats.charProfilesInitial[i].Stats[core.ER]
+
+		if char.Base.Key == core.Raiden {
+			stats.charSubstatFinal[i][core.ER] = 10
+		}
+
+		stats.charProfilesERBaseline[i].Stats[core.ER] += float64(stats.charSubstatFinal[i][core.ER]) * stats.substatValues[core.ER]
+	}
+
+	for i, char := range stats.charProfilesERBaseline {
+		if char.Base.Key != core.Raiden {
+			continue
+		}
+		opDebug = append(opDebug, "Raiden found in team comp - running secondary optimization routine...")
+		charDebug = stats.findOptimalERforChar(i, char, tolMean, tolSD)
+		opDebug = append(opDebug, charDebug...)
+	}
+
+	// Fix ER at previously found values then optimize all other substats
+	opDebug = append(opDebug, "Optimized ER Liquid Substats by character:")
+	printVal := ""
+	for i, char := range stats.charProfilesInitial {
+		printVal += fmt.Sprintf("%v: %.4g, ", char.Base.Key.String(), float64(stats.charSubstatFinal[i][core.ER])*stats.substatValues[core.ER])
+	}
+	opDebug = append(opDebug, printVal)
+
+	return opDebug
+}
+
+func (stats *SubstatOptimizerDetails) setCharProfilesCopy(charsToCopy []core.CharacterProfile) {
+	for i, char := range charsToCopy {
+		stats.charProfilesCopy[i] = char.Clone()
+	}
+}
+
+func (stats *SubstatOptimizerDetails) findOptimalERforChar(idxChar int, char core.CharacterProfile, tolMean float64, tolSD float64) []string {
+	var debug []string
+	var initialMean float64
+	var initialSD float64
+
+	debug = append(debug, fmt.Sprintf("%v", char.Base.Key))
+
+	for erStack := 0; erStack <= 10; erStack += 2 {
+		stats.charProfilesCopy[idxChar] = char.Clone()
+		stats.charProfilesCopy[idxChar].Stats[core.ER] -= float64(erStack) * stats.substatValues[core.ER]
+
+		stats.simcfg.Characters.Profile = stats.charProfilesCopy
+
+		result, _ := simulator.RunWithConfig(stats.cfg, stats.simcfg, stats.simopt)
+		debug = append(debug, fmt.Sprintf("%v: %v (%v)", stats.charSubstatFinal[idxChar][core.ER]-erStack, result.DPS.Mean, result.DPS.SD))
+
+		if erStack == 0 {
+			initialMean = result.DPS.Mean
+			initialSD = result.DPS.SD
+		}
+
+		condition := result.DPS.Mean/initialMean-1 < -tolMean || result.DPS.SD/initialSD-1 > tolSD
+		// For Raiden, we can't use DPS directly as a measure since she scales off of her own ER
+		// Instead we ONLY use the SD tolerance as big jumps indicate the rotation is becoming more unstable
+		if char.Base.Key == core.Raiden {
```

Comment:
i think this code here is more readable if rewritten as:
```go
if char.Base.Key == core.Raiden && result.DPS.SD/initialSD-1 > tolSD {
   break
}
```

as opposed to assigning to a variable condition, then checking if condition is true